projects/js-toolkit/packages/js-toolkit-core/src/transform/js/index.ts

```diff
@@ -121,8 +121,10 @@ async function mergeMaps(
 
 	const mergedMapGenerator = new SourceMapGenerator();
 
+	/* eslint-disable @typescript-eslint/await-thenable */
 	const oldMapConsumer = await new SourceMapConsumer(oldMap);
 	const newMapConsumer = await new SourceMapConsumer(newMap);
+	/* eslint-enable @typescript-eslint/await-thenable */
```

Comment:
What a horrible API... returns a Promise with the `new` operator 🤦‍♂️...

Seems like we might be able to avoid the ugly lint suppressions by using [`SourceMapConsumer.with()`](https://www.npmjs.com/package/source-map#sourcemapconsumerwith)?

Reply:
I think I tried may alternatives when I had to spend almost one day fighting with that, because:

1. As you say the API is horrible
2. Not happy with being horrible, they didn't make the correct typings for TS, so that nobody knew :facepalm: 

It is possible that `SourceMapConsumer.with()` fixes it (or not :sweat_smile: ) but I prefer not to try it in this PR. Also, we will remove this sooner or later, so maybe we don't even need to fix it ever.